s3/src/test/java/com/instaclustr/kafka/connect/s3/source/TopicPartitionSegmentParserTest.java

```diff
@@ -103,7 +105,7 @@ public void givenClosedStreamThrowIoException() throws IOException {
         InputStream nullInputStream = InputStream.nullInputStream();
         TopicPartitionSegmentParser topicPartitionSegmentParser = new TopicPartitionSegmentParser(nullInputStream, s3ObjectKey, "");
         nullInputStream.close();
-        Assert.expectThrows(IOException.class, () -> topicPartitionSegmentParser.getNextRecord(5L, TimeUnit.SECONDS));
+        Assert.expectThrows(ExecutionException.class, () -> topicPartitionSegmentParser.getNextRecord(5L, TimeUnit.SECONDS));
```

Comment:
This should now be checking if the thrown exception is actually wrapping an IOException as expected.
try{
//method call
} catch (Ex e){
//assert here the exception.getCause is an IOException  
}